packages/dx-grid-core/src/plugins/virtual-table-state/helpers.ts

```diff
@@ -43,12 +43,18 @@ export const mergeRows: MergeRowsFn = (
 };
 
 export const calculateRequestedRange: CalculateRequestedRangeFn = (
-  virtualRows, newRange, pageSize,
+  virtualRows, newRange, pageSize, referenceIndex,
 ) => {
   const loadedInterval = intervalUtil.getRowsInterval(virtualRows);
   const isAdjacentPage = Math.abs(loadedInterval.start - newRange.start) < 2 * pageSize;
   if (isAdjacentPage) {
-    return intervalUtil.difference(newRange, loadedInterval);
+    const calculatedRange = intervalUtil.difference(newRange, loadedInterval);
+    const firstHalfOfPage = calculatedRange.start - referenceIndex > pageSize / 2;
+    if (calculatedRange.start > referenceIndex && firstHalfOfPage) {
```

Comment:
Can we use only the next `calculatedRange.start - referenceIndex > pageSize / 2` codition?

if the above condition is true and `pageSize` > 0 this condition `calculatedRange.start > referenceIndex` is true